src/dspbase.jl

```diff
@@ -207,17 +207,48 @@ conv2(A::StridedMatrix{T}, B::StridedMatrix{T}) where {T<:Integer} =
 conv2(u::StridedVector{T}, v::StridedVector{T}, A::StridedMatrix{T}) where {T<:Integer} =
     round.(Int, conv2(float(u), float(v), float(A)))
 
+function check_padmode_kwarg(padmode::Symbol, su::Integer, sv::Integer)
+    if padmode == :default_longest
+        if su != sv
+            Base.depwarn(
+            """
+            The default behavior for xcorr will changed in future versions.
```

Comment:
couldn't hurt to be explicit here, and give a direct recommendation for what to do to keep things unchanged (for when you're updating old code): "the default pad mode of `xcorr` will be changed from `:longest` to `:none` in future versions. To keep current behavior specify `padmode=:longest`"

Reply:
done